inkscape: 0.92.5 -> 1.0

prusnak opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->

###### Motivation for this change

Update to Inkscape 1.0 final.

This PR also keeps Inkscape 0.92.5 as `inkscape_0` package, because with 1.0 the CLI interface has been changed completely. Packages using Inkscape 0.x CLI interface were patched to use this instead.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

prusnak commented:
During the packaging, I noticed that I need to add lots of dependencies that should have been dependencies of the libraries used, not Inkscape itself. Is it possible that these are missing from the `propagatedBuildInputs` of the packages used as dependencies?

I marked them explicitly with the comment and currently, these are:

```
+, pcre             # libpcre required by glib-2.0
+, libpthreadstubs  # pthread-stubs required by xcb
+, libXdmcp         # xdmcp required by xcb
+, utillinuxMinimal # mount required by gio-2.0
+, libselinux       # libselinux required by gio-2.0
+, libsepol         # libsepol required by libselinx
+, sqlite           # sqlite3 required by libsoup-2.4
+, libpsl           # libpsl required by libsoup-2.4
+, brotli           # libbrotlidec required by libsoup-2.4
+, icu              # icu-i18n required by libvisio-0.1, libcdr-0.1
+, libwpd           # libwpd-0.10 required by libwpg-0.3
+, libxkbcommon     # xkbcommon required by gdk-3.0
+, epoxy            # epoxy required by gdk-3.0
+, dbus             # dbus-1 required by atk-bridge-2.0
+, at-spi2-core     # atspi-2 required by atk-bridge-2.0
+, at-spi2-atk      # atspi-2 required by atk-bridge-2.0
+, libXtst          # xtst required by atspi-2
```

Can anyone advise? Maybe someone from the GTK packaging team? @raskin @vcunat @lethalman @worldofpeace

jtojnar commented:
See also https://github.com/NixOS/nixpkgs/pull/50286

prusnak commented:
I had to add quite a lot of extra dependencies to satisfy the pkg-config. (These are marked explicitly via the comment). Not sure what is happening here. I thought this was because `strictDeps = true`, but removing this line did not help.

Maybe someone from the GTK team has an idea what's going on? @raskin @vcunat @lethalman @worldofpeace

Here's the list of the extra dependencies I had to add:

```
    pcre             # libpcre required by glib-2.0
    libpthreadstubs  # pthread-stubs required by xcb
    libXdmcp         # xdmcp required by xcb
    utillinuxMinimal # mount required by gio-2.0
    libselinux       # libselinux required by gio-2.0
    libsepol         # libsepol required by libselinx
    sqlite           # sqlite3 required by libsoup-2.4
    libpsl           # libpsl required by libsoup-2.4
    brotli           # libbrotlidec required by libsoup-2.4
    icu              # icu-i18n required by libvisio-0.1, libcdr-0.1
    libwpd           # libwpd-0.10 required by libwpg-0.3
    libxkbcommon     # xkbcommon required by gdk-3.0
    epoxy            # epoxy required by gdk-3.0
    dbus             # dbus-1 required by atk-bridge-2.0
    at-spi2-core     # atspi-2 required by atk-bridge-2.0
    at-spi2-atk      # atspi-2 required by atk-bridge-2.0
    libXtst          # xtst required by atspi-2
```

prusnak commented:
Hm, I just checked master and I see the same pkg-config failures, so maybe these are expected and everything is OK?

worldofpeace commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -1,76 +1,91 @@
+    double-conversion
+    gdl
+    libsoup
+    pcre             # libpcre required by glib-2.0
+    libpthreadstubs  # pthread-stubs required by xcb
+    libXdmcp         # xdmcp required by xcb
+    utillinuxMinimal # mount required by gio-2.0
+    libselinux       # libselinux required by gio-2.0
+    libsepol         # libsepol required by libselinx
+    sqlite           # sqlite3 required by libsoup-2.4
+    libpsl           # libpsl required by libsoup-2.4
+    brotli           # libbrotlidec required by libsoup-2.4
+    icu              # icu-i18n required by libvisio-0.1, libcdr-0.1
+    libwpd           # libwpd-0.10 required by libwpg-0.3
+    libxkbcommon     # xkbcommon required by gdk-3.0
+    epoxy            # epoxy required by gdk-3.0
+    dbus             # dbus-1 required by atk-bridge-2.0
+    at-spi2-core     # atspi-2 required by atk-bridge-2.0
+    at-spi2-atk      # atspi-2 required by atk-bridge-2.0
+    libXtst          # xtst required by atspi-2
```
if these are in `Requires` for pkg-config, it will need them. I'm not sure all of these are propagatedBuildInputs in their respective packages, so that may by why you had to add them for pkgconfig to work.

prusnak replied:
My thoughts as well. That's why I prepared the full list, because I think these really belong into `propagatedBuildInputs` of the respective packages.

jtojnar replied:
I think those are just `Requires.private` and CMake complains about them but the build still continues.

worldofpeace commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -1,76 +1,87 @@
     gsl poppler imagemagick libwpg librevenge
     libvisio libcdr libexif potrace
-
+    double-conversion gdl libsoup
```
I'm seeing evaluation of nixpkgs-unstable here fail because `gdl` doesn't eval on darwin.
Does inkscape use it on macos? If yes, you should change `platforms` in that expression and see if we can get it to build.

prusnak replied:
I added gdl dependency to !darwin section only.

prusnak replied:
Done in b6870ea42b97a1e8c0dbedb72633424247a15f42

jtojnar commented:
For now, we should also bump to 0.92.5: https://github.com/NixOS/nixpkgs/pull/85184

prusnak committed:
inkscape: 0.92.5 -> 1.0

prusnak commented:
I rebased the PR on top of the current master (which now has inkscape 0.92.5 with some cleanup merged in).

I also removed the extra dependencies mentioned in the discussion above from the PR. I have them in a separate local commit in case these are still needed.

jtojnar commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -59,18 +63,19 @@ stdenv.mkDerivation rec {
     # the "Effects" menu.
     substituteInPlace src/extension/implementation/script.cpp \
-      --replace '"python-interpreter", "python"' '"python-interpreter", "${python3Env}/bin/python"'
+      --replace '"python3"' '"${python3Env}/bin/python"'
```
This will only work on Mac. I would prefer using patch from my PR as substitutions do not fail fast.

prusnak replied:
This also works for Linux, why do you think this works only for macOS?

jtojnar replied:
https://gitlab.com/inkscape/inkscape/-/blob/INKSCAPE_1_0_RC1/src/extension/implementation/script.cpp#L74-88

jtojnar replied:
Looks like they changed it up in https://gitlab.com/inkscape/inkscape/-/commit/7164e93e3c19bc206b73267eb3b0805c2e7597fa but that is not in the RC1.

prusnak replied:
Thanks, I'll fix this.

prusnak replied:
Addressed in  863becb

jtojnar commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -119,6 +129,8 @@ stdenv.mkDerivation rec {
     install_name_tool -change $out/lib/libinkscape_base.dylib $out/lib/inkscape/libinkscape_base.dylib $out/bin/inkview
   '';
 
+  ninjaFlags = [ "-v" ]; # make ninja verbose
```
Why this?

prusnak replied:
Don't we want the logs in the CI? If not, I'll remove this.

jtojnar replied:
I think the default verbosity should be sufficient. If we want this, we should consider doing it globally in ninja setup hook.

prusnak replied:
Addressed in  863becb

jtojnar commented on pkgs/applications/graphics/inkscape/default.nix:
```diff
@@ -106,9 +114,11 @@ stdenv.mkDerivation rec {
+    gtkspell3
   ] ++ stdenv.lib.optionals stdenv.isDarwin [
     cairo
+    gtk-mac-integration
```
Blegh, they still did not switch from `GtkApplication` and `GtkosxApplication` fully to `GApplication`.

prusnak replied:
:shrug:

prusnak committed:
gdl: enable on darwin

prusnak commented:
I addressed the comments from @jtojnar in 863becbf103

prusnak commented:
Darwin build timeouts after 1800 seconds

jtojnar commented:
Looks good, thanks.

jtojnar commented:
:tada:  https://inkscape.org/news/2020/05/04/introducing-inkscape-10/

minijackson commented:
So I wanted to test out Inkscape 1.0 and mechanically did a `nix-review` on this PR ^^

---

Result of `nix-review pr 85169` [1](https://github.com/Mic92/nix-review)
<details>
  <summary>10 package failed to build:</summary>

  - adapta-gtk-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - arx-libertatis (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - emojione (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-documents (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - mojave-gtk-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - numix-solarized-gtk-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - plata-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - ripcord (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - twemoji-color-font (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - wpa_supplicant_gui (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
</details>
<details>
  <summary>94 package were built:</summary>

  - aesop (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - arc-icon-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - arc-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - asciidoc-full (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - asciidoc-full-with-plugins (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - bibata-cursors (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - bibata-cursors-translucent (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - bibata-extra-cursors (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - bookworm (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - btrbk (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - capitaine-cursors (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - chrome-gnome-shell (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - clevis (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - dblatexFull (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - disorderfs (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - elementary-planner (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - faba-icon-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - faba-mono-icons (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - fim (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-control-center (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-session (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-shell (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-terminal (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.gnome-tweaks (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnome3.pomodoro (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - gnomeExtensions.gsconnect (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - hashit (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - inkscape (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - luksmeta (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - mate.mate-utils (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - moka-icon-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - monitor (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - nasc (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - notejot (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - numix-cursor-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.appcenter (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-calculator (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-calendar (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-camera (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-code (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-feedback (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-files (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-greeter (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-gsettings-schemas (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-icon-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-music (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-onboarding (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-photos (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-screenshot-tool (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-session-settings (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-shortcut-overlay (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-terminal (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.elementary-videos (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.gala (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.sideload (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-a11y (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-about (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-applications (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-bluetooth (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-datetime (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-display (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-keyboard (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-mouse-touchpad (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-network (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-notifications (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-onlineaccounts (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-pantheon-shell (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-power (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-printers (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-security-privacy (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-sharing (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-plug-sound (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.switchboard-with-plugs (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-applications-menu (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-bluetooth (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-datetime (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-keyboard (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-network (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-nightlight (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-notifications (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-power (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-session (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-indicator-sound (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pantheon.wingpanel-with-indicators (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - pop-gtk-theme (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - quilter (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - regextester (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - snippetpixie (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - spice-up (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - tang (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - udiskie (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
  - vocal (gnome3.gnome-tweak-tool ,gnome3.gnome_control_center ,gnome3.gnome_session ,gnome3.gnome_shell ,gnome3.gnome_terminal)
</details>

---

From a quick glance it seems due to the removal of the `--without-gui` / `-z` CLI option.

Here are the logs of failed derivations, I hope it is useful:

https://gist.github.com/minijackson/46b4913eddd3cddecce09337179b2f62

ripcord's log is empty for some reason.

prusnak commented:
I reworked the PR from 1.0RC1 to 1.0 final, but I am unable to build locally because of the following error:

```
trying https://media.inkscape.org/dl/resources/file/inkscape-1.0.tar.xz
curl: (22) The requested URL returned error: 503
Warning: Transient problem: HTTP error Will retry in 1 seconds. 3 retries
```

Maybe the Inkscape servers are overloaded, or there is another problem.

I will try again tomorrow.

jtojnar commented:
Yes `inkscape` has completely revamped their CLI breaking many themes, cc @romildo https://gitlab.com/inkscape/inbox/-/issues/2416

prusnak commented:
Breaking CLI is a major bummer. What we can do is the following:

1. build inkscape 0.92.5 without gui as a new derivation `inkscape_cli`
2. require `inkscape_cli` instead of `inkscape` in packages
3. `inkscape` derivation will be the latest 1.0

What do you think @jtojnar?

7c6f434c commented:
>1. build inkscape 0.92.5 without gui as a new derivation `inkscape_cli`

Maybe `inkscape_cli_0x` ?

prusnak committed:
inkscape -> inkscape_0

We can't update inkscape to 1.0 without keeping 0.x in tree,
because the CLI interface has been changed and lots of packages
use this interface for SVG conversion.

prusnak committed:
treewide: replace inkscape with inkscape_0

The Inkscape CLI interface has been changed with 1.x and packages
are not adjusted yet to this change. Let's use inkscape 0.x instead.

prusnak commented:
I was not able to figure out how to build Inkscape without the GUI (I am not sure it's even possible), so I went ahead and kept the 0.x version as `inkscape_0`. I also replaced all dependencies with this new package, hopefully fixing all build breakages.

jtojnar commented:
Since there is only ten packages affected, I would prefer patching them.

- [ ] adapta-gtk-theme
- [ ] arx-libertatis https://github.com/arx/ArxLibertatis/pull/256
- [ ] emojione
- [ ] gnome3.gnome-documents
- [ ] mojave-gtk-theme
- [ ] numix-solarized-gtk-theme
- [ ] plata-theme
- [ ] ripcord
- [ ] twemoji-color-font https://github.com/eosrei/twemoji-color-font/issues/75
- [ ] wpa_supplicant_gui

prusnak commented:
> Since there is only ten packages affected, I would prefer patching them.

I agree this would be a better option. However, I think there are some packages missing from your list:

- [ ] bibata-cursors
- [ ] bibata-extra-cursors
- [ ] bibata-cursors-translucent
- [ ] arc-theme
- [ ] capitaine-cursors
- [ ] numix-cursor-theme
- [ ] pop-gtk-theme
- [ ] mate-utils
- [ ] elementary-icon-theme
- [ ] fim
- [ ] dblatex

Infinisil commented:
```
hash mismatch in fixed-output derivation '/nix/store/g8svbgzzw4ha3wygc1sfkrci74kxnpdb-inkscape-1.0.tar.xz':
  wanted: sha256:0ci4fg5c3phis9dqh5xddnlsc05lca6gj0pzyhc46x41fafavpg0
  got:    sha256:1fwl7yjkykqb86555k4fm24inhc40mrvxqwgl2v2vi9alv8j7hc9
```

prusnak commented:
@Infinisil fixed in e9bc6e3

aaronjanse commented:
Hmm, icons seems to be acting weird. On my laptop (i3, no desktop environment):

inkscape 92.4 (build from `master@{2 days ago}`): everything normal  
inkscape 92.5: crashes immediately (`Gtk::IconThemeError`)  
inkscape 1.0: starts but all icons are blank (but otherwise inkscape looks beautiful)

I have `gnome3.adwaita-icon-theme` in my `environment.systemPackages`.

spacekookie commented:
I just ran inkscape from this PR and noticed a crash when trying to open the file-picker window (to open a file):

```
(org.inkscape.Inkscape:7308): GLib-GIO-ERROR **: 01:49:57.209: No GSettings schemas are installed on the system
```

prusnak commented:
This seems similar to https://github.com/NixOS/nixpkgs/pull/87457 - @jtojnar can you confirm that moving `glib` to `nativeBuildInputs` is correct way of fixing the issue?

jtojnar commented:
Maybe try adding it there, it should still remain in `buildInputs`

prusnak commented:
@spacekookie can you try adding `glib` to `nativeBuildInputs` to see whether this helps?

jtojnar commented:
Guix actually managed to unbundle the dependencies: https://lists.gnu.org/archive/html/guix-patches/2020-05/msg00365.html

Pablo1107 commented:
I import this package from @prusnak fork tarball and it works perfect on NixOS 20.03.

jtojnar commented:
Rebased (fixing the conflicts) and added `glib` to `nativeBuildInputs`. It looks like it adds `$GSETTINGS_SCHEMAS_PATH` to the build environment at least.

raboof commented:
@GrahamcOfBorg build inkscape

raboof commented:
@GrahamcOfBorg build gdl, gdl.passthru.tests, inkscape, inkscape.passthru.tests

raboof commented:
@GrahamcOfBorg build gdl gdl.passthru.tests inkscape inkscape.passthru.tests

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/what-are-your-goals-for-20-09/8035/7

7c6f434c reviewed (approved):
Code looks good to me

worldofpeace commented on pkgs/applications/graphics/inkscape/0.x.nix:
```diff
@@ -0,0 +1,133 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Vector graphics editor";
```
can we note this is the old version?

prusnak replied:
Addressed in 7eb4a0ae274c193c4b7be629f602e5781100d9d1 - changed the wording to "Legacy version of vector graphics editor"

prusnak commented:
Rebased on top of the current master + changed the description of inkscape_0 package to "Legacy version of vector graphics editor".

worldofpeace reviewed (approved):
LGTM, haven't tested running it though.

worldofpeace commented:
I'm also fine with keeping around the old version of the package because of the cli changes.

aaronjanse commented:
Ty all! I'm super excited to use this!
